boards/kernel_layout.ld

```diff
@@ -337,6 +337,40 @@ SECTIONS
     } > ram
     _eappmem = ORIGIN(ram) + LENGTH(ram);
 
+
+    /* Place attributes at the end of the kernel ROM region. These will be used
+     * for host-side tools to learn about the installed kernel.
+     *
+     * Attributes are stored as TLVs, but going backwards in flash.
+     *
+     *                           2 byte   2 byte   4 byte
+     * -----------+------------+--------+--------+------------+
+     *            | Value...   | Type   | Length | TOCK       | Start of `prog`
+     * -----------+------------+--------+--------+------------+
+     * <-TLVs...-> <-----TLV--------------------> <-Sentinel->
+     *
+     * `Length` specifies the size of the value (i.e. not include the header).
+     */
+    .attributes :
+    {
+        . = ABSOLUTE(ORIGIN(rom) + LENGTH(rom) - 16);
+
+        /* TLV: App Memory
+         * This indicates the start address and size of RAM available for
+         * userspace apps.
+         */
+        LONG(_sappmem) /* Address of start of app memory. */
+        LONG(_eappmem - _sappmem) /* Length of app memory. */
+        SHORT(0x0101) /* Type = App RAM Region = 0x0101 */
+        SHORT(8)      /* Length = 8 bytes */
+
+        BYTE(84) /* T */
+        BYTE(79) /* O */
+        BYTE(67) /* C */
+        BYTE(75) /* K */
+    }  > rom
```

Comment:
(line of code irrelevant) Do you have a sense for how bad the cost is in terms of time? Maybe just for whatever board you were testing with at least? Presumably it scales with the size of Flash we have reserved for the kernel.

Reply:
Well for jlink it doesn't really change anything.

For serial, on hail-master:

```
make program
    Finished release [optimized + debuginfo] target(s) in 0.09s
   text	   data	    bss	    dec	    hex	filename
 112324	      0	  19932	 132256	  204a0	/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail
30d94f325136ae6a83cfe968f9a30f2ad4026502991ce8fbe0513b9f4b352299	/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail.bin
tockloader  flash --address 0x10000 /Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail.bin
[INFO   ] No device name specified. Using default name "tock".
[INFO   ] Using "/dev/cu.usbserial-c098e513000a - Hail IoT Module - TockOS - Hail IoT Module - TockOS".
[STATUS ] Flashing binary to board...
[INFO   ] Padding binary with 320 0xFFs.
100%|█████████████████████████████████████████████████| 220/220 [00:14<00:00, 15.63it/s]
[INFO   ] CRC check passed. Binaries successfully loaded.
[INFO   ] Finished in 16.800 seconds
```

And with this PR:

```
make program
    Finished release [optimized + debuginfo] target(s) in 0.10s
   text	   data	    bss	    dec	    hex	filename
 112036	  19040	  19932	 151008	  24de0	/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail
afea7a7e09427cb4799a9f7a480d2c6ea8161dc62fc67d724de768c2cb3d3f45	/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail.bin
tockloader  flash --address 0x10000 /Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail.bin
[INFO   ] No device name specified. Using default name "tock".
[INFO   ] Using "/dev/cu.usbserial-c098e513000a - Hail IoT Module - TockOS - Hail IoT Module - TockOS".
[STATUS ] Flashing binary to board...
100%|█████████████████████████████████████████████████| 256/256 [00:16<00:00, 15.63it/s]
[INFO   ] CRC check passed. Binaries successfully loaded.
[INFO   ] Finished in 19.128 seconds
```

So, a little longer.